pyvista/plotting/mapper.py

```diff
@@ -103,6 +103,11 @@ def set_scalars(
                 scalar_bar_args.setdefault('n_labels', 0)
                 _using_labels = True
 
+            # Use only the real component if an array is complex
+            if np.issubdtype(scalars.dtype, complex):
+                scalars = scalars.astype(float)
```

Comment:
Don't we want to raise an informative warning of our own instead of relying on NumPy's generic warning?

Reply:
I like how it matches `matplotlib`'s behavior. Users can google it and get way more informative examples rather than having to interpret our warning.

If you disagree, I can change it.

Reply:
Hmm, good point, having good googlability is probably a good idea. Let's leave it like this, we can always introduce our own warning later if we get issues about the current state.